compiler/damlc/tests/daml-test-files/RecordDotUpdates.daml

```diff
@@ -0,0 +1,50 @@
+-- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its affiliates.
```

Comment:
Looks like a great set of tests, a few missing case:
1. Nested field puns
2. errors on typos in field names. In particular interested in locations there. Maybe better to test that in the shake tests, those already have a set of tests that test source locations.

Reply:
I think I already got nested field puns in the last line of that file. Adding an error location test in the shake tests.

Reply:
```
y {text, inner.text = text, int, inner.int = int}
```
I only see non-nested field puns here. Am I looking at the wrong line?

Reply:
Added the line `z = let Outer {..} = y in Outer {..}` to the tests.